PR: https://github.com/odoo/odoo/pull/253079

From: 3259a7fae4ca4ee98f7e02da5109c43cd9e77f69
From: Tatiana Skvortsova (taskv)
Date: 2026-04-13 06:36:16

Breaking data model changes scores: del:3 + add:0, change matches:
-    departure_do_cancel_time_off_requests = fields.Boolean(related='version_id.departure_do_cancel_time_off_requests',
-    do_cancel_time_off_requests = fields.Boolean(
-    departure_do_cancel_time_off_requests = fields.Boolean(related='departure_id.do_cancel_time_off_requests')

Total Changes: 25

[IMP] l10n_be_hr_payroll: Improve End of Collaboration UX

Improve the UX of the "End of Collaboration" procedure, particularly for Belgium.

The option to select which actions are executed upon employee departure
(delete user, archive employee, unassign car, etc.) has been removed.

All actions are now executed automatically. However, it is still possible
to define the date on which they will be applied.

Related PRs:
Enterprise: odoo/enterprise#110147
Upgrade: odoo/upgrade#9653

task-5958676

test

closes odoo/odoo#253079

Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_holidays/models/hr_employee.py
+++ b/addons/hr_holidays/models/hr_employee.py
@@ -47,9 +47,6 @@ class HrEmployee(models.Model):
         ('presence_holiday_present', 'Present but on leave')])
     member_of_department = fields.Boolean('Member of Department', compute='_compute_member_of_department', search='_search_part_of_department')
 
-    departure_do_cancel_time_off_requests = fields.Boolean(related='version_id.departure_do_cancel_time_off_requests',
-        inherited=True, readonly=False, groups="hr.group_hr_user")
-
     def _compute_current_work_entry_type_id(self):
         self.current_work_entry_type_id = False
 

--- a/addons/hr_holidays/models/hr_employee_departure.py
+++ b/addons/hr_holidays/models/hr_employee_departure.py
@@ -2,33 +2,27 @@
 
 from datetime import timedelta
 
-from odoo import fields, models
+from odoo import models
 
 
 class HrEmployeeDeparture(models.Model):
     _inherit = 'hr.employee.departure'
 
-    do_cancel_time_off_requests = fields.Boolean(
-        string="Cancel Time Off Requests",
-        default=True,
-        help="Set the running allocations validity's end and delete future time off.")
-
     def action_register(self):
         super().action_register()
-        cancel_leave_departures = self.filtered(lambda d: d.do_cancel_time_off_requests)
-        if not cancel_leave_departures:
+        if not self:
             return
         all_leaves_sudo = self.sudo().env['hr.leave'].search([
-            ('employee_id', 'in', cancel_leave_departures.employee_id.ids),
-            ('date_to', '>', min(cancel_leave_departures.mapped('departure_date'))),
+            ('employee_id', 'in', self.employee_id.ids),
+            ('date_to', '>', min(self.mapped('departure_date'))),
         ])
         all_allocations_sudo = self.sudo().env['hr.leave.allocation'].search([
-            ('employee_id', 'in', cancel_leave_departures.employee_id.ids),
+            ('employee_id', 'in', self.employee_id.ids),
             '|',
                 ('date_to', '=', False),
-                ('date_to', '>', min(cancel_leave_departures.mapped('departure_date'))),
+                ('date_to', '>', min(self.mapped('departure_date'))),
         ])
-        for departure in cancel_leave_departures:
+        for departure in self:
             employee_leaves_sudo = all_leaves_sudo.filtered_domain([
                 ('employee_id', '=', departure.employee_id.id),
                 ('date_to', '>', departure.departure_date),

--- a/addons/hr_holidays/models/hr_version.py
+++ b/addons/hr_holidays/models/hr_version.py
@@ -19,8 +19,6 @@ class HrVersion(models.Model):
     _inherit = 'hr.version'
     _description = 'Employee Contract'
 
-    departure_do_cancel_time_off_requests = fields.Boolean(related='departure_id.do_cancel_time_off_requests')
-
     @api.constrains('contract_date_start', 'contract_date_end')
     def _check_contracts(self):
         self._get_leaves()._check_contracts()
